src/ExpressionEvaluator/CSharp/Test/ResultProvider/ExpansionTests.cs

```diff
@@ -69,6 +69,66 @@ public void Primitives()
             Verify(FormatResult("\"\"", CreateDkmClrValue("")), EvalResult("\"\"", "\"\"", "string", "\"\"", DkmEvaluationResultFlags.RawString, editableValue: "\"\""));
         }
 
+        [Fact, WorkItem(40179, "https://github.com/dotnet/roslyn/issues/40179")]
+        public void UnderscoreMembersComeFirst()
+        {
+            var source =
+@"class C
+{
+    C Z;
+    object _z;
+    C A;
+    string _a;
+    object _;
+}";
+            var assembly = GetAssembly(source);
+            var type = assembly.GetType("C");
+            var rootExpr = "new C()";
+            var value = CreateDkmClrValue(Activator.CreateInstance(type));
+            var evalResult = FormatResult(rootExpr, value);
+            Verify(evalResult,
+                EvalResult(rootExpr, "{C}", "C", rootExpr, DkmEvaluationResultFlags.Expandable));
+            var children = GetChildren(evalResult);
+            Verify(children,
+                EvalResult("@_", "null", "object", "(new C()).@_", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("_a", "null", "string", "(new C())._a", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("_z", "null", "object", "(new C())._z", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("A", "null", "C", "(new C()).A", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("Z", "null", "C", "(new C()).Z", DkmEvaluationResultFlags.CanFavorite));
+        }
+
+        [Fact, WorkItem(40179, "https://github.com/dotnet/roslyn/issues/40179")]
+        public void LetterUnderscoreMembersComeSecond()
+        {
+            var source =
+@"class C
+{
+    C Z;
+    object _z;
+    object m_z;
+    C A;
+    string _a;
+    object _;
+    string m_a;
+}";
+            var assembly = GetAssembly(source);
+            var type = assembly.GetType("C");
+            var rootExpr = "new C()";
+            var value = CreateDkmClrValue(Activator.CreateInstance(type));
+            var evalResult = FormatResult(rootExpr, value);
+            Verify(evalResult,
+                EvalResult(rootExpr, "{C}", "C", rootExpr, DkmEvaluationResultFlags.Expandable));
+            var children = GetChildren(evalResult);
+            Verify(children,
+                EvalResult("@_", "null", "object", "(new C()).@_", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("_a", "null", "string", "(new C())._a", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("_z", "null", "object", "(new C())._z", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("m_a", "null", "string", "(new C()).m_a", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("m_z", "null", "object", "(new C()).m_z", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("A", "null", "C", "(new C()).A", DkmEvaluationResultFlags.CanFavorite),
+                EvalResult("Z", "null", "C", "(new C()).Z", DkmEvaluationResultFlags.CanFavorite));
+        }
```

Comment:
>} [](start = 8, length = 1)

Are we testing a mix of fields and properties with `_` names?

Reply:
Since the change doesn't have any logic pertaining to that anymore, I'm not sure it's relevant. I could add...